Islandora/islandora_scholar
# Islandora 2251

**JIRA Ticket**: https://jira.duraspace.org/browse/ISLANDORA-2251

https://docs.google.com/spreadsheets/d/1PtK16Fo2q4jTMgz17DYmV1nauvoGDEi6f2fhl9K1ZIA

# What does this Pull Request do?

Makes the XPaths editable through an admin GUI

![ISLANDORA SCHOLAR XPATH CONFIGURATION](https://user-images.githubusercontent.com/2738244/41784378-8481d2c8-760d-11e8-935e-527b6c5c02c2.png)

# What's new?

Islandora Scholar XPath Configuration Section located on the /admin/islandora/solution_pack_config/scholar page. It allows you to modify the previously hard coded XPaths and reset them at any time. 

# How should this be tested?
* Start islandora_vagrant and ingest an item (prior to switching to this PR)
  * Inspect it and test that the search results look like expects
  * from the terminal
```terminal
      $ vagrant ssh
      $ cd /var/ww/drupal
      $ drush vget islandora_scholar_default_search_xpath
      No matching variable found.
```
* Switch to this PR
  * Go to /admin/islandora/solution_pack_config/scholar page and click 'Save Configuration'
  * from the terminal
```terminal
      $ vagrant ssh
      $ cd /var/ww/drupal
      $ drush vget islandora_scholar_default_search_xpath
...
```
  * This time you should see several results
  * Ingest an item
  * Try modifying an XPath (easiest is to switch title with something else)
  * Check that the change shows up on a new ingest and an existing one. 
  * Also try adding new values to the citation_form.xml or thesis_form.xml and see if pointing to the new values works. 

# Additional Notes:

* This change requires documentation to be updated.
* This should not impact existing installations.
* This does **NOT** modify the submission form.
* This allows for the 52 XPaths and 3 subpaths to be set
* This exclude subpaths in a few spots where they occurred.
    * Check google doc for excluded some paths

# Interested parties
Tag (@ mention) interested parties or, if unsure, @Islandora/7-x-1-x-committers
